django-hatstall/hatstall/views.py

```diff
@@ -124,22 +127,32 @@ def identity(request, identity_id):
 @login_required
 def update_enrollment(request, identity_id, organization):
     """
-    Update profile enrollment dates
-    It first removes old enrollment
-    and creates a new one (base on the new dates)
+    Update the enrollment from a profile updating dates
+    using `Enroll` command, merging overlapping dates.
     """
-    err = None
     if not Conf.check_conf():
         return redirect('shdb')
     if request.method != 'POST':
         return redirect('identities list')
-    db = sortinghat_db_conn()
-    old_start_date = parser.parse(request.POST.get('old_start_date'))
-    old_end_date = parser.parse(request.POST.get('old_end_date'))
+
+    db_args = {
+        "user": Conf.shdb_user,
+        "password": Conf.shdb_pass,
+        "database": Conf.shdb_name,
+        "host": Conf.shdb_host,
+        "port": Conf.shdb_port
+    }
+
     start_date = parser.parse(request.POST.get('start_date'))
     end_date = parser.parse(request.POST.get('end_date'))
-    sortinghat.api.delete_enrollment(db, identity_id, organization, old_start_date, old_end_date)
-    sortinghat.api.add_enrollment(db, identity_id, organization, start_date, end_date)
+
+    enroll_cmd = Enroll(**db_args)
+    enroll_cmd.enroll(identity_id,
+                      organization,
+                      from_date=start_date,
+                      to_date=end_date,
+                      merge=True)
```

Comment:
After performing some more tests, I saw there is one case which could be problematic having this option activated:

* We add an enrollment. It has default dates (1900-2100).
* Now, we set the `start` date to `2018-01-01`.
* Then, we set that `start` date to `2017-12-31` (1 day before).

If we try to set `2018-01-01` again, the old date is preserved `2017-12-31`, because the dates from those enrollments get merged into the wider range (in this case, the one with the older date). 

Should we preserve this `merge` option or should we remove it so the user can manage the different date ranges manually?

Reply:
On this case, the user should either edit the enrollment or remove the entry and add a new one.